docs/user-guide/expressions/user-defined-functions.md

```diff
@@ -50,112 +20,114 @@ Use cases for `map_batches` in the `group_by` context are slim. They are only us
+--8<-- "python/user-guide/expressions/user-defined-functions.py:diff_from_mean_numba"
 ```
 
-### Combining multiple column values
+## Missing data is not allowed when calling generalized ufuncs
 
-If we want to have access to values of different columns in a single `map_elements` function call, we can create `struct` data
-type. This data type collects those columns as fields in the `struct`. So if we'd create a struct from the columns
-`"keys"` and `"values"`, we would get the following struct elements:
+Before being passed to a user-defined function like `diff_from_mean_numba()`, a `Series` will be converted to a NumPy array.
+Unfortunately, NumPy arrays don't have a concept of missing data.
+If there is missing data in the original `Series`, this means the resulting array won't actually match the `Series`.
 
-```python
-[
-    {"keys": "a", "values": 10},
-    {"keys": "a", "values": 7},
-    {"keys": "b", "values": 1},
-]
-```
+If you're calculating results item by item, this doesn't matter.
+For example, `numpy.log()` gets called on each individual value separately, so those missing values don't change the calculation.
+But if the result of a user-defined function depend on multiple values in the `Series`, it's not clear what exactly should happen with the missing values.
 
-In Python, those would be passed as `dict` to the calling Python function and can thus be indexed by `field: str`. In Rust, you'll get a `Series` with the `Struct` type. The fields of the struct can then be indexed and downcast.
+Therefore, when calling generalized ufuncs such as Numba functions decorated with `@guvectorize`, Polars will raise an error if you try to pass in a `Series` with missing data.
+How do you get rid of missing data?
+Either [fill it in](missing-data.md) or [drop it](https://docs.pola.rs/py-polars/html/reference/dataframe/api/polars.DataFrame.drop_nulls.html) before calling your custom function.
+
+## Combining multiple column values
+
+If you want to pass multiple columns to a user-defined function, you can use `Struct`s, which are [covered in detail in a different section](structs.md).
+The basic idea is to combine multiple columns into a `Struct`, and then the function can extract the columns back out:
 
 {{code_block('user-guide/expressions/user-defined-functions','combine',[])}}
 
 ```python exec="on" result="text" session="user-guide/udf"
 --8<-- "python/user-guide/expressions/user-defined-functions.py:combine"
 ```
 
-`Structs` are covered in detail in the next section.
+## Streaming calculations
 
-### Return types?
+Passing the full `Series` to the user-defined function has a cost: it may use a lot of memory, as its contents are copied into a NumPy array.
+You can use a `is_elementwise=True` argument to [:material-api: `map_batches`](https://docs.pola.rs/py-polars/html/reference/expressions/api/polars.Expr.map_batches.html) to stream results into the function, which means it might not get all values at once.
```

Comment:
```suggestion
You can use the `is_elementwise=True` argument to [:material-api: `map_batches`](https://docs.pola.rs/py-polars/html/reference/expressions/api/polars.Expr.map_batches.html) to stream results into the function, which means it might not get all values at once.
```